amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableRuntime.java

```diff
@@ -409,7 +409,7 @@ private boolean updateConfigInternal(Map<String, String> properties) {
   }
 
   public void addTaskQuota(TaskRuntime.TaskQuota taskQuota) {
-    doAs(OptimizingMapper.class, mapper -> mapper.insertTaskQuota(taskQuota));
+    doAsAlone(OptimizingMapper.class, mapper -> mapper.insertTaskQuota(taskQuota));
```

Comment:
It seems we started another transaction to update the quota.
Another method to solve this issue is using the same transaction and ignoring the persistent error. We can add a method like `doAsIgnoreError` in PersistentBase to get that.

What do you think is the better method? @rfyu

Reply:
thanks for your suggestion. It is acceptable to handle it using the same transaction. I will make the necessary modifications accordingly.